### ISSUE TITLE
We need a simplified dev workflow document for newcomers

### ISSUE THREAD

**Issue by moorepants** [sympy/sympy#14022]
I find https://github.com/sympy/sympy/wiki/Development-workflow to be an extremely overwhelming document. It likely doesn't help us get new folks on board by sending them straight to this.

My hunch is that we could create an additional development workflow guide targeted at newcomers that gave an absolute bare minimum set of information for them to get a basic pull request in. Then our current dev workflow guide would be what they learn after doing the this one. It should not have any optional information but simply one simple path for them to get an issue and submit a PR. It could be called something like "Five simple steps to submitting your first pull request".

**Comment by normalhuman** [comment]
Would the five simple steps be: 

1. [Pick an issue to fix](https://github.com/sympy/sympy/wiki/Development-workflow#pick-an-issue-to-fix)
1. [Create a git branch](https://github.com/sympy/sympy/wiki/Development-workflow#create-a-new-branch) with your fix.
1. [Run the tests](https://github.com/sympy/sympy/wiki/Development-workflow#be-sure-that-all-tests-of-sympy-pass)
1. [Commit the changes](https://github.com/sympy/sympy/wiki/Development-workflow#commit-the-changes)
1. [Write a pull request](https://github.com/sympy/sympy/wiki/Development-workflow#create-a-patch-file-or-pull-request-for-github)

I think these are readable articles. So I don't see why the "5 steps" could not be just a highlighted paragraph on top of the existing file.

It may seem overwhelming when read sequentially, with virtual environments and tuning bash prompts, but the parts that matter most are well written. (For context: my git knowledge is at the level of [xkcd 1597](https://xkcd.com/1597/), and found this guide to be very clear and helpful.)

**Comment by moorepants** [comment]
If they are well written I think this should be pulled out into a separate document so that they aren't clouded by all the other information. It may also be possible to remove the git branch step. I wonder if having them submit a PR entirely from the Github interface would remove a ton of the detailed Git steps.

Great comic, btw :)

**Comment by moorepants** [comment]
Also, you could remove the run the tests step because Travis can run it...

**Comment by ethankward** [comment]
It's still a good idea to run the tests before you commit so you aren't wasting Travis resources.

**Comment by moorepants** [comment]
@ethankward I totally agree, but I'm trying to envision the absolute minimum step process for someone to submit a pull request to SymPy. Running the tests locally is definitely a very good idea, but is not essential. I think the entry point we have for the development workflow is not scaffolded for incremental learning. It is all presented in one huge document.

**Comment by moorepants** [comment]
Here is a shot at a different five steps:

1. Identify and choose an "easy to fix" issue.
2. Find the related `test_*.py` files and the related source files in the `sympy` directory on Github.
3. Click the edit button on the test file in Github and write a unit test function that demonstrates the desired behavior.
4. Commit the changes with an informational commit message and select create a pull request.
5. Wait for the tests to run on Travis and for feedback on the pull request.

That's is far as I could get with just 5, but it would at least get someone to a good starting point. Maybe editing the test and source files could be steps 3 and 4.

**Comment by bjodah** [comment]
I like the idea. I someone has the time I think a narrated screencast hosted on e.g. YouTube could be useful. Regarding the text I think that words like "commit", "pull request" & "Travis" are not necessarily obvious to a newcomer.

**Comment by ylemkimon** [comment]
GitHub provides a great guide on the development workflow: https://guides.github.com/introduction/flow/ and video guides: https://www.youtube.com/githubguides.
